### ISSUE TITLE
edition compatibility: No warning emitted for crate/macro of the same name

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/rust#56684]
[Originally reported upstream](https://github.com/rust-lang-nursery/rustfix/issues/153) this code:

```rust
#![warn(rust_2018_compatibility)]
#![allow(unused_imports)]

#[macro_use]
extern crate toml; // 0.4.8

mod x {
    use toml;
}
```

[when compiled emits no warnings](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=48aef9a186ae2726f22b61029b001f12), but [when compiled in the 2018 edition fails to compile](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=814800927bc6c820c8db23363fa78b5f).

This appears to have something to do with the fact that the `toml` crate has a `toml` macro perhaps? It seems unclear!

**Comment by ehuss** [comment]
I think this can be closed. This was fixed in 1.32 via #56759.
